recipe-server/client/control/components/ExtensionList.js

```diff
@@ -0,0 +1,66 @@
+import React, { PropTypes as pt } from 'react';
+import { connect } from 'react-redux';
+import { browserHistory } from 'react-router';
+import { Table, Thead, Th, Tr, Td } from 'reactable';
+
+import QueryExtensions from './data/QueryExtensions';
+import { getAllExtensions } from '../state/extensions/selectors';
+import { isRequestInProgress } from '../state/requests/selectors';
+
+
+class ExtensionList extends React.Component {
+  static propTypes = {
+    extensions: pt.array.isRequired,
+    isLoading: pt.bool.isRequired,
+  };
+
+  showDetails(extension) {
+    browserHistory.push(`/control/extension/${extension.get('id')}/`);
+  }
+
+  render() {
+    const { extensions, isLoading } = this.props;
+
+    return (
+      <div>
+        <QueryExtensions />
+        <div className="fluid-8">
+          {
+            isLoading &&
+              <div
+                className="loading callout"
+                children={'Loading...'}
+              />
+          }
+
+          <Table
+            className="recipe-list"
+            sortable
+          >
+            <Thead>
+              <Th column="name">
+                <span>Name</span>
+              </Th>
+            </Thead>
+            {extensions.map(extension =>
+              <Tr
+                key={extension.get('id')}
+                onClick={() => this.showDetails(extension)}
```

Comment:
This creates an anonymous function once per render per extension. We probably instead want the row to be it's own component with the click handler as a method on the component. That'd avoid re-renders as well, since the anonymous functions don't compare as equal.